core/src/SYCL/Kokkos_SYCL_Instance.cpp

```diff
@@ -42,6 +42,7 @@
 //@HEADER
 */
 
+#include <Kokkos_Core.hpp>
```

Comment:
Why are you including `<Kokkos_Core.hpp>`?

Reply:
This is for `kokkos_malloc`. I could try forward declaring in `Kokkos_core_fwd.hpp` instead.

Reply:
It just makes little sense to include everything and below include selected headers.